### PR TITLE
[DATA-FRAME] adds specs and yml tests for existing endpoints

### PR DESCRIPTION
**Pull request by benwtrent**
This adds specs and some yml tests for our _data_frame endpoints.

YML tests are not all encompassing, more will probably have to be written in the future. Additionally, docs are still a TODO.

### CODE REVIEW

File: x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.put_data_frame_transform.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "data_frame.put_data_frame_transform": {
+    "documentation": "TODO",
+    "methods": [ "PUT" ],
+    "url": {
+      "path": "/_data_frame/transforms/{transform_id}",
+      "paths": [ "/_data_frame/transforms/{transform_id}", "/_data_frame/transforms" ],
```
Comment (droberts195) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.put_data_frame_transform.json]: `"/_data_frame/transforms"` is not supported on the REST layer.  The ID has to be in the URL.

File: x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.put_data_frame_transform.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      "parts": {
+        "transform_id": {
+          "type": "string",
+          "required": false,
+          "description": "The id of the new transform. If not provided, must be included in the body"
```
Comment (droberts195) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.put_data_frame_transform.json]: > If not provided, must be included in the body

Should be:

The id may also optionally be provided in the body, but must match the id in the URL

(Or alternatively we could just omit this as a minor detail - what does put_job say about this?  The reason for this functionality is to make it easier to take config returned from the GET endpoint and send it back unaltered to the PUT endpoint.)

File: x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.get_data_frame_transform.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "documentation": "TODO",
+    "methods": [ "GET" ],
+    "url": {
+      "path": "/_data_frame/transforms/{transform_id}",
+      "paths": [ "/_data_frame/transforms/{transform_id}"],
+      "parts": {
+        "transform_id": {
+          "type": "string",
+          "required": false,
+          "description": "The id of the transforms to get, '*' implies get all transforms"
```
Comment (droberts195) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.get_data_frame_transform.json]: How about `_all`?  If that's a synonym for `*` maybe we should mention that too.

If that's not implemented then that's an inconsistency with other APIs that we should fix at some point.  Don't fix it as part of this PR but raise an issue so someone can add `_all` in the future.

File: x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.get_data_frame_transform_stats.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "documentation": "TODO",
+    "methods": [ "GET" ],
+    "url": {
+      "path": "/_data_frame/transforms/{transform_id}/_stats",
+      "paths": [ "/_data_frame/transforms/{transform_id}/_stats", "/_data_frame/transforms/_stats" ],
+      "parts": {
+        "transform_id": {
+          "type": "string",
+          "required": false,
+          "description": "The id of the transform for which to get stats. Empty or * implies all transforms"
```
Comment (droberts195) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.get_data_frame_transform_stats.json]: What about `_all`?

**Review by droberts195 — COMMENTED** [review]
Thanks for picking this up!

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/docbldesx

**Comment by benwtrent** [comment]
run elasticsearch-ci/docbldesx

**Pull request merged**